cmd/smith/app/app.go

```diff
@@ -74,13 +74,14 @@ type App struct {
 }
 
 func (a *App) Run(ctx context.Context) error {
+	defer a.Logger.Sync()
 	// Plugins
 	pluginContainers, err := a.loadPlugins()
 	if err != nil {
 		return err
 	}
 	for pluginName := range pluginContainers {
-		log.Printf("Loaded plugin: %q", pluginName)
+		a.Logger.Sugar().Infof("Loaded plugin: %q", pluginName)
```

Comment:
every call on `.Sugar()` creates a new instance of logger, so probably to invoke it only once.
See https://github.com/uber-go/zap/blob/master/logger.go#L126
and https://github.com/uber-go/zap#quick-start

Reply:
Also I don't think the performance of logging is critical for us. Can't we just always use SugaredLogger?

Reply:
Yes, performance is not critical, so I call `Sugar()` whenever I feel like 😛 
I like that with structured logging (without sugar) I can use helper functions - see `logz` package. How would I do the same with sugared logger?

Reply:
Also the sugar call is very cheap, so I don't bother keeping the sugared logger around. That is actually how I started, but then sugared logger does not have the nice typed api.